OpenRA.Mods.Cnc/Activities/LayMines.cs

```diff
@@ -56,50 +64,51 @@ public override bool Tick(Actor self)
 			if (IsCanceling)
 				return true;
 
-			if (rearmableInfo != null && ammoPools.Any(p => p.Info.Name == minelayer.Info.AmmoPoolName && !p.HasAmmo()))
+			if ((minefield == null || minefield.Contains(self.Location)) && CanLayMine(self, self.Location))
 			{
-				// Rearm (and possibly repair) at rearm building, then back out here to refill the minefield some more
-				rearmTarget = self.World.Actors.Where(a => self.Owner.Stances[a.Owner] == Stance.Ally
-					&& rearmableInfo.RearmActors.Contains(a.Info.Name))
-					.ClosestTo(self);
-
-				if (rearmTarget == null)
-					return true;
-
-				// Add a CloseEnough range of 512 to the Rearm/Repair activities in order to ensure that we're at the host actor
-				QueueChild(new MoveAdjacentTo(self, Target.FromActor(rearmTarget)));
-				QueueChild(movement.MoveTo(self.World.Map.CellContaining(rearmTarget.CenterPosition), rearmTarget));
-				QueueChild(new Resupply(self, rearmTarget, new WDist(512)));
-				returnToBase = true;
-				return false;
-			}
+				if (rearmableInfo != null && ammoPools.Any(p => p.Info.Name == minelayer.Info.AmmoPoolName && !p.HasAmmo()))
+				{
+					// Rearm (and possibly repair) at rearm building, then back out here to refill the minefield some more
+					rearmTarget = self.World.Actors.Where(a => self.Owner.Stances[a.Owner] == Stance.Ally
+					                                           && rearmableInfo.RearmActors.Contains(a.Info.Name))
```

Comment:
The indentation here really doesn't look correct.

Reply:
Fixed.